pkg/virt-controller/watch/ovm_test.go

```diff
@@ -0,0 +1,260 @@
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/testutils"
+)
+
+var _ = Describe("OfflineVirtualMachine", func() {
+
+	Context("One valid OfflineVirtualMachine controller given", func() {
+
+		var ctrl *gomock.Controller
+		var vmInterface *kubecli.MockVMInterface
+		var ovmInterface *kubecli.MockOfflineVirtualMachineInterface
+		var vmSource *framework.FakeControllerSource
+		var ovmSource *framework.FakeControllerSource
+		var vmInformer cache.SharedIndexInformer
+		var ovmInformer cache.SharedIndexInformer
+		var stop chan struct{}
+		var controller *OVMController
+		var recorder *record.FakeRecorder
+		var mockQueue *testutils.MockWorkQueue
+		var vmFeeder *testutils.VirtualMachineFeeder
+
+		syncCaches := func(stop chan struct{}) {
+			go vmInformer.Run(stop)
+			go ovmInformer.Run(stop)
+			Expect(cache.WaitForCacheSync(stop, vmInformer.HasSynced, ovmInformer.HasSynced)).To(BeTrue())
+		}
+
+		BeforeEach(func() {
+			stop = make(chan struct{})
+			ctrl = gomock.NewController(GinkgoT())
+			virtClient := kubecli.NewMockKubevirtClient(ctrl)
+			vmInterface = kubecli.NewMockVMInterface(ctrl)
+			ovmInterface = kubecli.NewMockOfflineVirtualMachineInterface(ctrl)
+
+			vmInformer, vmSource = testutils.NewFakeInformerFor(&v1.VirtualMachine{})
+			ovmInformer, ovmSource = testutils.NewFakeInformerFor(&v1.OfflineVirtualMachine{})
+			recorder = record.NewFakeRecorder(100)
+
+			controller = NewOVMController(vmInformer, ovmInformer, recorder, virtClient)
+			// Wrap our workqueue to have a way to detect when we are done processing updates
+			mockQueue = testutils.NewMockWorkQueue(controller.Queue)
+			controller.Queue = mockQueue
+			vmFeeder = testutils.NewVirtualMachineFeeder(mockQueue, vmSource)
+
+			// Set up mock client
+			virtClient.EXPECT().VM(v12.NamespaceDefault).Return(vmInterface).AnyTimes()
+			virtClient.EXPECT().OfflineVirtualMachine(v12.NamespaceDefault).Return(ovmInterface).AnyTimes()
+		})
+
+		addOfflineVirtualMachine := func(ovm *v1.OfflineVirtualMachine) {
+			syncCaches(stop)
+			mockQueue.ExpectAdds(1)
+			ovmSource.Add(ovm)
+			mockQueue.Wait()
+		}
+
+		It("should create missing VM", func() {
+			ovm, vm := DefaultOVM(true)
+
+			addOfflineVirtualMachine(ovm)
+
+			// expect creation called
+			vmInterface.EXPECT().Create(gomock.Any()).Do(func(arg interface{}) {
+				Expect(arg.(*v1.VirtualMachine).ObjectMeta.Name).To(Equal("testvm"))
+			}).Return(vm, nil)
+
+			// expect update status is called
+			ovmInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
+				Expect(arg.(*v1.OfflineVirtualMachine).Status.Conditions[0].Type).To(Equal(v1.OfflineVirtualMachineRunning))
+			}).Return(ovm, nil)
+
+			controller.Execute()
+
+			testutils.ExpectEvent(recorder, SuccessfulCreateVirtualMachineReason)
+		})
+
+		It("should delete VM when stopped", func() {
+			ovm, vm := DefaultOVM(false)
+
+			addOfflineVirtualMachine(ovm)
+
+			// Add VM to the cache
```

Comment:
Is that a left-over or there for documentation? I would tend to remove the commented out code ...